src/button/button-internals.js

```diff
@@ -15,6 +16,22 @@ import {getOverrides} from '../helpers/overrides.js';
 
 import type {ButtonPropsT} from './types.js';
 
+function RenderEnhancer(props) {
+  const {Enhancer, ...restProps} = props;
+  if (
```

Comment:
Why not just to check for `isValidElementType` and delete this first branch?

Reply:
Fixed it up so it now only handles strings early. This is required because `'div'` is a valid element type and can be called like `<'div' {...props />`. This provides a good example of why we should never allow `React.Node` as a render prop type and default to `React.AbstractComponent`